### ISSUE TITLE
Bell icon on Android platform

### ISSUE THREAD

**Issue by patrickboulay** [katzer/cordova-plugin-local-notifications#1308]
## Your Environment

* Plugin version: master branch, latest version (May 31th)
* Platform: Android
* OS version: 5.1.1
* Device manufacturer / model: Samsung Grand Prime
* Cordova version: 7.0.1
* Cordova platform version : 
Installed platforms:
  android 6.2.3
  ios 4.4.0
* Plugin config
                    cordova.plugins.notification.local.schedule({
                        id: app.notificationId,
                        title: "text",
                        text: "text",
                    });

## Expected Behavior

When a notification pop, I should have my application icon in the notification bar.

## Actual Behavior

On Android, I have a bell icon, I dont know where it come from.  On iOS, I see my default application.

## Steps to Reproduce

Schedule a notification.

## Context

On going in background mode or when my application is in background.

## Debug logs

The log file is attached to this issue.
[log.txt.gz](https://github.com/katzer/cordova-plugin-local-notifications/files/1043467/log.txt.gz)

**Comment by Tawpie** [comment]
The bell is Android's default icon, the bell is used when the operating system can't find a *correct* icon file. By correct, the icon files (there must be one file for each screen density), must be in the correctly named folder, must be the exact size specified and must be images that conform to Android's requirements.

Google for the Android requirements.

Android is very very very picky about this and if anything at all doesn't conform to requirements, Android will use the bell.

(spelling and grammer powered by autocorrupt)

**Comment by rwillett** [comment]
@tawpie is spot on here. 

I also note that there is nothing in your code that actually puts your icon in place, so the system is working as designed.

**Comment by patrickboulay** [comment]
I though when the icon parameter was not passed, it will take my app icon
by default..

I tried some parameters and its not working..

in my config.xml, I have:
    <platform name="android">
        <icon density="ldpi" src="res/icons/android/36.png" />
        <icon density="mdpi" src="res/icons/android/48.png" />
        <icon density="hdpi" src="res/icons/android/72.png" />
        <icon density="xhdpi" src="res/icons/android/96.png" />
        <icon density="xxhdpi" src="res/icons/android/144.png" />
        <icon density="xxxhdpi" src="res/icons/android/192.png" />
    </platform>

and in my platform/android folders(generated by cordova), I have:
res/mipmap-hdpi/icon.png
res/mipmap-ldpi/icon.png
res/mipmap-mdpi/icon.png
res/mipmap-xhdpi/icon.png
res/mipmap-xxhdpi/icon.png
res/mipmap-xxxhdpi/icon.png


My icons seems to be good because my app has

What should be the parameter?
icon: 'res://icon.png' ??
icon: 'res://icon' ??

No idea..

Thanks for the help.

**Comment by Tawpie** [comment]
@patricboulay — you’re really close. Try adding smallicon: “res://icon” AND put your various icon files in drawable-hdpi, drawable-mpdi etc. Cordova generates the mipmap folders, but our mipmap folders do NOT have our notification icons in them, only the app icons. Our notification icons are in drawable- folders.

Of course, I’m assuming the icons are properly sized and grayscale with transparency per https://developer.android.com/guide/practices/ui_guidelines/icon_design_action_bar.html

BTW, it’s a terminology detail, but the bar at the top of the screen with the carrier and clock and battery etc. is called the “Action Bar” in android and the “Status Bar” in iOS. I’m assuming that’s what you meant by “notification bar”.

**Comment by rwillett** [comment]
@patrickboulay Did this work?

**Comment by MarcBalaban** [comment]
I believe android must have tightened up it's requirements recently.  This problem also appeared for me in the last few weeks.

I've changed the icon size to 48x48 and set the icon colour scheme to black and white, but still getting the default bell icon.

**Comment by patrickboulay** [comment]
I didnt try, Im waiting about my friend to give me the icon for that.. I
let you know.

Pat

**Comment by Tawpie** [comment]
@MarcBalaban You need all 4 icons: 18x18, 24x24, 36x36 and 48x48, each being a png with an alpha channel (include transparency). Those icons must be in platforms/android/res/drawable-ldpi, -mdpi, -hdpi and -xhdpi. Try that.

**Comment by rwillett** [comment]
Just to add to this, some editors are not 100% accurate with getting the Alpha channel transparent. It's worth checking this. 

From memory the Affinity products work, but there are online checkers as well. getting the Android icons right is hard work (for some reason) and you need to have it all right.

**Comment by patrickboulay** [comment]
I still didnt check the Android icon but I remark a problem on iOS side. On
an iPhone 6S plus, I saw the default Cordova icon.
This is my config.xml:
    <platform name="ios">
        <icon height="180" src="res/icons/ios/Icon-60@3x.png" width="180" />
        <icon height="60" src="res/icons/ios/Icon-60.png" width="60" />
        <icon height="120" src="res/icons/ios/Icon-60@2x.png" width="120" />
        <icon height="76" src="res/icons/ios/Icon-76.png" width="76" />
        <icon height="152" src="res/icons/ios/Icon-76@2x.png" width="152" />
        <icon height="40" src="res/icons/ios/Icon-Small-40.png" width="40"
/>
        <icon height="80" src="res/icons/ios/Icon-Small-40@2x.png"
width="80" />
        <icon height="57" src="res/icons/ios/Icon.png" width="57" />
        <icon height="114" src="res/icons/ios/Icon@2x.png" width="114" />
        <icon height="72" src="res/icons/ios/Icon-72.png" width="72" />
        <icon height="144" src="res/icons/ios/Icon-72@2x.png" width="144" />
        <icon height="29" src="res/icons/ios/Icon-Small.png" width="29" />
        <icon height="58" src="res/icons/ios/Icon-Small@2x.png" width="58"
/>
        <icon height="50" src="res/icons/ios/Icon-Small-50.png" width="50"
/>
        <icon height="100" src="res/icons/ios/Icon-Small-50@2x.png"
width="100" />
    </platform>

In the cordova documention, the 6s plus should use the 180x180 icon.

Do im doing something wrong?

Pat

**Comment by patrickboulay** [comment]
Alright, works fine on Android now!! Thanks a lot for you help!

But still have problem with the iPhone 6S notification icon. Does it have
something special for this?

Pat

**Comment by rohitkawale** [comment]
I have a not.png file in android res folder

$cordovaLocalNotification.add({
                    id: "$"+data.id,
                    date: alarmTime2,
                    message: "1 day left for your favorite show to Begin!",
                    title: data.name+" "+alarmTime,
                    autoCancel: true,
                    sound: "file://sounds/reminder.mp3",
                    icon: "res://not.png",
                    smallicon: "res://not.png",
                    data:data.id
                }).then(function () {
                    //alert("The notification for one day before has been set from dynamic");
                });


i'm getting a bell icon

**Comment by anuj-njoshi** [comment]
how to pass  large icon parameter

**Comment by marcovtwout** [comment]
In addition to the solution in https://github.com/katzer/cordova-plugin-local-notifications/issues/1308#issuecomment-305680542

If you are using a build service like Ionic Pro you cannot add the icons to platforms/android directly. However, you can make Cordova copy the files from a source folder by configuring your icons in config.xml with [`<resource-file>`:](https://cordova.apache.org/docs/en/latest/plugin_ref/spec.html#resource-file)
```
<platform name="android">
    ...
    <resource-file src="resources/android/notification-icon/icon-mdpi.png" target="res/drawable-mdpi/my_notification_icon.png" />
    <resource-file src="resources/android/notification-icon/icon-hdpi.png" target="res/drawable-hdpi/my_notification_icon.png" />
    <resource-file src="resources/android/notification-icon/icon-xhdpi.png" target="res/drawable-xhdpi/my_notification_icon.png" />
    <resource-file src="resources/android/notification-icon/icon-xxhdpi.png" target="res/drawable-xxhdpi/my_notification_icon.png" />
    <resource-file src="resources/android/notification-icon/icon-xxxhdpi.png" target="res/drawable-xxxhdpi/my_notification_icon.png" />
</platform>
```

And you set the notification option like this (without extension):

```
smallIcon: "res://my_notification_icon",
```

**Comment by katzer** [comment]
That should be solved now. A small icon (`smallIcon`) is required on Android, the big icon (`icon`) is optional.

**Comment by brunosfreitas** [comment]
As @marcovtwout pointed out that was the only solution that worked for me. I've edited my `config.xml` adding all the `<resource-file>` references to the new folder `resources/android/notification-icon`. On this new folder i've put all the pngs with multiple resolutions. That solved the problem for the (` small icon `) but now that grayscale icon is on the Action bar and the notification panel.

I had to do the same thing with the (` icon `) because it used the same images as the (` small icon `) and i needed a different one (with colors). As so my ` config.xml ` is the same as @marcovtwout listed but with a second roll of icons for ` icon ` entry.
